src/events/Windows.cpp

```diff
@@ -687,7 +687,7 @@ void Events::listener_unmapWindow(void* owner, void* data) {
     g_pHyprOpenGL->makeWindowSnapshot(PWINDOW);
 
     // swallowing
-    if (PWINDOW->m_pSwallowed && g_pCompositor->windowExists(PWINDOW->m_pSwallowed)) {
+    if (PWINDOW->m_pSwallowed && g_pCompositor->windowExists(PWINDOW->m_pSwallowed) && !PWINDOW->m_bReadyToDelete) {
```

Comment:
I dont think the last condition will ever be false

Reply:
Yeah, I messed up